src/components/DatatableV2/Table/components/TableCheckbox.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { forwardRef } from 'react';
+import cls from 'classnames';
+
+import * as checked from '../../../../theme/icons/check';
+import * as indeterminate from '../../../../theme/icons/minus';
+import { Checkbox } from '../../../forms';
+import { TableCheckboxProps } from './TableCheckbox.types';
+
+const generateIconProps = ({ width, height, svgPathData }) => ({
+  viewBox: `0 0 ${width} ${height}`,
+  children: <path d={svgPathData} fill="currentColor" />,
+});
+
+const Mark = ({ children, ...props }) => (
+  <svg role="img" xmlns="http://www.w3.org/2000/svg" {...props}>
+    {children}
+  </svg>
+);
+
+const TableCheckbox = forwardRef(
+  (
+    { name, checkboxId, isIndeterminate, ...props }: TableCheckboxProps,
+    ref: React.MutableRefObject<HTMLInputElement>,
+  ): React.ReactElement => (
+    <div className="ds-table-checkbox">
```

Comment:
All those CSS styles will come later or I'm missing something?

Reply:
It will come soon, probably in next PR

Reply:
You used CSS modules instead of styled-components because of performance?

Reply:
correct